pip/locations.py

```diff
@@ -187,6 +187,8 @@ def distutils_scheme(dist_name, user=False, home=None, root=None,
     i.user = user or i.user
     if user:
         i.prefix = ""
+    else:
+        i.prefix = prefix or i.prefix
```

Comment:
Just like in distutils, I think pip should raise if the user provides both `--user` and `--prefix`.